src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const totalArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (stateCopy[key]) {
+            delete stateCopy[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(stateCopy)) {
+          delete stateCopy[key];
+        }
```

Comment:
```suggestion
        stateCopy = {}
```
in this task also possible just reassign stateCopy on empty object